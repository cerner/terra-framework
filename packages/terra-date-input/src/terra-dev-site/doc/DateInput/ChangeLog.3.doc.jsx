import React from 'react';
import DocTemplate from 'terra-doc-template';
import ChangeLog from '../../../../CHANGELOG.md';

const DocPage = () => (
  <DocTemplate
    packageName="date-input"
    srcPath="https://github.com/cerner/terra-framework/tree/master/packages/date-input"
    readme={ChangeLog}
  />
);

export default DocPage;
