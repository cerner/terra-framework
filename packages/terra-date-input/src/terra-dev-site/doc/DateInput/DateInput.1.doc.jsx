/* eslint-disable import/no-webpack-loader-syntax, import/first, import/extensions, import/no-duplicates, import/no-unresolved */
import React from 'react';
import DocTemplate from 'terra-doc-template';
import ReadMe from '../../../../docs/README.md';
import { name } from '../../../../package.json';

// Component Source
import DateInputSrc from '!raw-loader!../../../../src/DateInput';

// Example Files
import DefaultDateInput from '../example/DefaultDateInput';
import DefaultDateInputSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/DefaultDateInput.jsx';
import MonthDayYearDateInput from '../example/MonthDayYearDateInput';
import MonthDayYearDateInputSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/MonthDayYearDateInput.jsx';
import DayMonthYearDateInput from '../example/DayMonthYearDateInput';
import DayMonthYearDateInputSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/DayMonthYearDateInput.jsx';

const DocPage = () => (
  <DocTemplate
    packageName={name}
    readme={ReadMe}
    srcPath={`https://github.com/cerner/terra-framework/tree/master/packages/${name}`}
    examples={[
      {
        title: 'Default DateInput',
        example: <DefaultDateInput />,
        source: DefaultDateInputSrc,
      },
      {
        title: 'Month Day Year Display Format DateInput',
        example: <MonthDayYearDateInput />,
        source: MonthDayYearDateInputSrc,
      },
      {
        title: 'Day Month Year Display Format DateInput',
        example: <DayMonthYearDateInput />,
        source: DayMonthYearDateInputSrc,
      },
    ]}
    propsTables={[
      {
        componentName: 'DateInput',
        componentSrc: DateInputSrc,
      },
    ]}
  />
);

export default DocPage;
