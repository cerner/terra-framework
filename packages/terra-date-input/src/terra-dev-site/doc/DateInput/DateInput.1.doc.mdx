import { Badge } from 'terra-date-input/package.json?dev-site-package';

import DateInputPropsTable from 'terra-date-input/src/DateInput?dev-site-props-table';

import DefaultDateInput from '../example/DefaultDateInput?dev-site-example';
import MonthDayYearDateInput from '../example/MonthDayYearDateInput?dev-site-example';
import DayMonthYearDateInput from '../example/DayMonthYearDateInput?dev-site-example';

<Badge />

# Terra Date Input

The DateInput allows for easy data entry of known dates like birthdays, anniversaries, etc. The display of the DateInput is localized based on the locale but can be customized via the `displayFormat` prop. The DateInput uses the ISO 8601 format for date values (YYYY-MM-DD).

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-date-input`

<!-- AUTO-GENERATED-CONTENT:START Peer Dependencies -->
## Peer Dependencies

This component requires the following peer dependencies be installed in your app for the component to properly function.

| Peer Dependency | Version |
|-|-|
| react | ^16.8.5 |
| react-dom | ^16.8.5 |
| react-intl | ^2.8.0 |

<!-- AUTO-GENERATED-CONTENT:END -->

## Component Features

 * [Cross-Browser Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#cross-browser-support)
 * [Mobile Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#mobile-support)
 * [Internationalization Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#internationalization-i18n-support)
 * [LTR/RTL Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#ltr--rtl-support)

[View component source code](https://github.com/cerner/terra-framework/tree/master/packages/terra-date-input)

## Examples
<DefaultDateInput title= 'Default DateInput' />
<MonthDayYearDateInput title= 'Month Day Year Display Format DateInput' />
<DayMonthYearDateInput title='Day Month Year Display Format DateInput' />

## Date Input props table
<DateInputPropsTable /> 
