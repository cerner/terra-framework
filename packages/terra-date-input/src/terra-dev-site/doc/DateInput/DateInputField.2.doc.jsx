/* eslint-disable import/no-webpack-loader-syntax, import/first, import/extensions, import/no-duplicates, import/no-unresolved */
import React from 'react';
import DocTemplate from 'terra-doc-template';
import ReadMe from '../../../../docs/DateInputField.md';
import { name } from '../../../../package.json';

// Component Source
import DateInputFieldSrc from '!raw-loader!../../../../src/DateInputField';

// Example Files
import DefaultDateInputField from '../example/DefaultDateInputField';
import DefaultDateInputFieldSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/DefaultDateInputField.jsx';
import InlineDateInputField from '../example/InlineDateInputField';
import InlineDateInputFieldSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/InlineDateInputField.jsx';
import OptionalDateInputField from '../example/OptionalDateInputField';
import OptionalDateInputFieldSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/OptionalDateInputField.jsx';
import RequiredDateInputField from '../example/RequiredDateInputField';
import RequiredDateInputFieldSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/RequiredDateInputField.jsx';
import HideRequiredDateInputField from '../example/HideRequiredDateInputField';
import HideRequiredDateInputFieldSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/HideRequiredDateInputField.jsx';
import DayMonthYearDateInputField from '../example/DayMonthYearDateInputField';
import DayMonthYearDateInputFieldSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/DayMonthYearDateInputField.jsx';
import MonthDayYearDateInputField from '../example/MonthDayYearDateInputField';
import MonthDayYearDateInputFieldSrc from '!raw-loader!../../../../src/terra-dev-site/doc/example/MonthDayYearDateInputField.jsx';

const DocPage = () => (
  <DocTemplate
    packageName={name}
    readme={ReadMe}
    srcPath={`https://github.com/cerner/terra-framework/tree/master/packages/${name}`}
    examples={[
      {
        title: 'Default DateInputField',
        example: <DefaultDateInputField />,
        source: DefaultDateInputFieldSrc,
      },
      {
        title: 'Required DateInputField',
        example: <RequiredDateInputField />,
        source: RequiredDateInputFieldSrc,
      },
      {
        title: 'Hide Required DateInputField',
        example: <HideRequiredDateInputField />,
        source: HideRequiredDateInputFieldSrc,
      },
      {
        title: 'Inline DateInputField',
        example: <InlineDateInputField />,
        source: InlineDateInputFieldSrc,
      },
      {
        title: 'Optional DateInputField',
        example: <OptionalDateInputField />,
        source: OptionalDateInputFieldSrc,
      },
      {
        title: 'Month Day Year Display Format DateInputField',
        example: <MonthDayYearDateInputField />,
        source: MonthDayYearDateInputFieldSrc,
      },
      {
        title: 'Day Month Year Display Format DateInputField',
        example: <DayMonthYearDateInputField />,
        source: DayMonthYearDateInputFieldSrc,
      },
    ]}
    propsTables={[
      {
        componentName: 'DateInputField',
        componentSrc: DateInputFieldSrc,
      },
    ]}
  />
);

export default DocPage;
