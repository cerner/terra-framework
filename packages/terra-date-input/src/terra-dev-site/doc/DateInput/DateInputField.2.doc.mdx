import { Badge } from 'terra-date-input/package.json?dev-site-package';

import DateInputFieldPropsTable from 'terra-date-input/src/DateInputField?dev-site-props-table';

import DefaultDateInputField from '../example/DefaultDateInputField?dev-site-example';
import InlineDateInputField from '../example/InlineDateInputField?dev-site-example';
import OptionalDateInputField from '../example/OptionalDateInputField?dev-site-example';
import RequiredDateInputField from '../example/RequiredDateInputField?dev-site-example';
import HideRequiredDateInputField from '../example/HideRequiredDateInputField?dev-site-example';
import DayMonthYearDateInputField from '../example/DayMonthYearDateInputField?dev-site-example';
import MonthDayYearDateInputField from '../example/MonthDayYearDateInputField?dev-site-example';
import IncompleteDateInputField from '../example/IncompleteDateInputField?dev-site-example';

<Badge />

# Terra Date Input Field

DateInputField is a component of terra-date-input that provides a container for rendering the related date inputs. DateInputField includes displays for help text, error text, required or optional indication, and invalid states.

## Component Features

 * [Cross-Browser Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#cross-browser-support)
 * [Mobile Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#mobile-support)
 * [Internationalization Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#internationalization-i18n-support)
 * [LTR/RTL Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#ltr--rtl-support)

[View component source code](https://github.com/cerner/terra-framework/tree/master/packages/terra-date-input)

## Examples
<DefaultDateInputField title='Default DateInputField' />
<RequiredDateInputField title='Required DateInputField' />
<HideRequiredDateInputField  title='Hide Required DateInputField' />
<InlineDateInputField title='Inline DateInputField'/>
<OptionalDateInputField title='Optional DateInputField' />
<MonthDayYearDateInputField title='Month Day Year Display Format DateInputField' />
<DayMonthYearDateInputField title='Day Month Year Display Format DateInputField' />
<IncompleteDateInputField title='Incomplete DateInputField' description='Applies theme-specific styling for incomplete. ***Note: Only use incomplete if given specific guidance, reserved for specific applications when no value has been provided. Not for general use.***' />

## Date input field props table
<DateInputFieldPropsTable />
