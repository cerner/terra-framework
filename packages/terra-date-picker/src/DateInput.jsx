import React from 'react';
import PropTypes from 'prop-types';
import Button from 'terra-button';
import IconCalendar from 'terra-icon/lib/icon/IconCalendar';
import Input from 'terra-form-input';
import DateUtil from './DateUtil';
import styles from './DatePicker.module.scss';

const Icon = <IconCalendar />;

const propTypes = {
  /**
   * Callback ref to pass into the calendar button dom element.
   */
  buttonRefCallback: PropTypes.func,
  /**
   * Custom input attributes to apply to the date input.
   */
  // eslint-disable-next-line react/forbid-prop-types
  inputAttributes: PropTypes.object,
  /**
   * Name of the date input.
   */
  name: PropTypes.string,
  /**
   * A callback function triggered when the input or calendar button loses focus.
   */
  onBlur: PropTypes.func,
  /**
   * A callback function to execute when a valid date is selected or entered.
   */
  onChange: PropTypes.func,
  /**
   * The onInputClick callback function from react-datepicker to show the picker when clicked.
   */
  onClick: PropTypes.func,
  /**
   * A callback function triggered when the date input receives focus.
   */
  onFocus: PropTypes.func,
  /**
   * A callback function triggered when the calendar button receives focus.
   */
  onButtonFocus: PropTypes.func,
  /**
   * The onInputKeyDown callback function from react-datepicker to handle keyboard navigation.
   */
  onKeyDown: PropTypes.func,
  /**
   * The placeholder text to display in the date input.
   */
  placeholder: PropTypes.string,
  /**
   * A callback function to let the containing component (e.g. modal) to regain focus.
   */
  releaseFocus: PropTypes.func,
  /**
   * A callback function to request focus from the containing component (e.g. modal).
   */
  requestFocus: PropTypes.func,
  /**
   * @private Internal prop for showing date picker.
   */
  shouldShowPicker: PropTypes.bool,
  /**
   * The selected or entered date value to display in the date input.
   */
  value: PropTypes.string,
};

const defaultProps = {
  buttonRefCallback: undefined,
  inputAttributes: undefined,
  name: undefined,
  onBlur: undefined,
  onChange: undefined,
  onClick: undefined,
  onFocus: undefined,
  onButtonFocus: undefined,
  onKeyDown: undefined,
  placeholder: undefined,
  releaseFocus: undefined,
  requestFocus: undefined,
  value: undefined,
};

const contextTypes = {
  /* eslint-disable consistent-return */
  intl: (context) => {
    if (context.intl === undefined) {
      return new Error('Component is internationalized, and must be wrapped in terra-base');
    }
  },
};

// eslint-disable-next-line react/prefer-stateless-function
class DatePickerInput extends React.Component {
  constructor(props) {
    super(props);

    this.handleOnButtonClick = this.handleOnButtonClick.bind(this);
    this.handleOnChange = this.handleOnChange.bind(this);
    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (this.shouldShowPicker && !prevProps.shouldShowPicker && this.props.onClick) {
      this.props.onClick();
    }
  }

  handleOnButtonClick(event) {
    // The picker is about to display so request focus from the containing component (e.g. modal) if it has the focus trapped.
    if (this.props.requestFocus) {
      this.props.requestFocus();
    }

    const attributes = Object.assign({}, this.props.inputAttributes);

    if (!attributes.readOnly && this.onCalendarButtonClick && this.props.onClick) {
      this.onCalendarButtonClick(event, this.props.onClick);
    }
  }

  handleOnChange(event) {
    if (!DateUtil.validDateInput(event.target.value)) {
      return;
    }

    if (this.props.onChange) {
      this.props.onChange(event);
    }
  }

  handleOnKeyDown(event) {
    // The picker will be dismissed if one of these keys is pressed and the focus will be released so release the focus to the containing component.
    if (this.props.releaseFocus && (event.key === 'Enter' || event.key === 'Escape' || event.key === 'Tab')) {
      this.props.releaseFocus();

      if (event.key === 'Tab') {
        event.preventDefault();
      }
    }

    if (this.props.onKeyDown) {
      this.props.onKeyDown(event);
    }
  }

  render() {
    const {
      buttonRefCallback,
      inputAttributes,
      name,
      onBlur,
      onChange,
      onClick,
      onFocus,
      onButtonFocus,
      onKeyDown,
      placeholder,
      releaseFocus,
      requestFocus,
      value,
      ...customProps
    } = this.props;

    this.onCalendarButtonClick = customProps.onCalendarButtonClick;
    this.shouldShowPicker = customProps.shouldShowPicker;

    delete customProps.onCalendarButtonClick;
    delete customProps.shouldShowPicker;

    const additionalInputProps = Object.assign({}, customProps, inputAttributes);

    const dateValue = DateUtil.convertToISO8601(value, DateUtil.getFormatByLocale(this.context.intl.locale));
    const buttonText = this.context.intl.formatMessage({ id: 'Terra.datePicker.openCalendar' });

    return (
      <div className={styles['custom-input']}>
        <input
          // Create a hidden input for storing the name and value attributes to use when submitting the form.
          // The data stored in the value attribute will be the visible date in the date input but in ISO 8601 format.
          data-terra-date-input-hidden
          type="hidden"
          name={name}
          value={dateValue}
        />
        <Input
          {...additionalInputProps}
          className={styles.input}
          type="text"
          name={'terra-date-'.concat(name)}
          value={value}
          onChange={this.handleOnChange}
          placeholder={placeholder}
          onFocus={onFocus}
          onBlur={onBlur}
        />
        <Button
          className={styles.button}
          text={buttonText}
          onClick={this.handleOnButtonClick}
          onKeyDown={this.handleOnKeyDown}
          icon={Icon}
          isIconOnly
          isCompact
          isDisabled={additionalInputProps.disabled}
          onBlur={onBlur}
          onFocus={onButtonFocus}
          refCallback={buttonRefCallback}
        />
      </div>
    );
  }
}

DatePickerInput.propTypes = propTypes;
DatePickerInput.defaultProps = defaultProps;
DatePickerInput.contextTypes = contextTypes;

export default DatePickerInput;
