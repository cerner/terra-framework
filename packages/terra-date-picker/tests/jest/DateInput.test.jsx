import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import { renderWithIntl } from 'terra-enzyme-intl';
import DateInput from '../../lib/DateInput';

it('should render a default date input', () => {
  const dateInput = renderWithIntl(<DateInput />);
  expect(dateInput).toMatchSnapshot();
});

it('should render a default date input with all props', () => {
  const dateInput = renderWithIntl((
    <DateInput
      inputAttributes={{ id: 'terra-date-input' }}
      name="date-input"
      placeholder="MM/DD/YYYY"
      value="01/01/2017"
      onChange={() => {}}
      onClick={() => {}}
      onKeyDown={() => {}}
    />
  ));

  expect(dateInput).toMatchSnapshot();
});

it('throws error on missing locale prop in Base', () => {
  global.console = { error: jest.fn() };

  expect(() => {
    render(<DateInput name="date-input" />);
  }).toThrowError();
  // eslint-disable-next-line no-console
  expect(console.error).toBeCalledWith(expect.stringContaining('Component is internationalized, and must be wrapped in terra-base'));
});
