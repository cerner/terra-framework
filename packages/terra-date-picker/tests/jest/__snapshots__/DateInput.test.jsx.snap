// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass in refCallback as the ref prop of the calendar button 1`] = `
<DatePickerInput
  buttonRefCallback={
    [MockFunction] {
      "calls": Array [
        Array [
          <button
            aria-disabled="false"
            aria-label="Terra.datePicker.openCalendar"
            class="button neutral compact button"
            title="Terra.datePicker.openCalendar"
            type="button"
          >
            <span
              class="button-label icon-only"
            >
              <span
                class="icon"
              >
                <svg
                  aria-hidden="true"
                  class="tui-Icon icon icon-svg"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 48 48"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M43 6h-4V2.2C39 1 38 0 36.8 0h-1.7C33.9.1 33 1 33 2.2V6H15V2.2C15 1 14 0 12.8 0h-1.7C9.9.1 9 1 9 2.2V6H5c-1.1 0-2 .9-2 2v38c0 1.1.9 2 2 2h24c1.3-.1 2.5-.6 3.4-1.4l11.2-11.1c.8-.9 1.3-2.1 1.4-3.4V8c0-1.1-.9-2-2-2zM6 21h36v10H30c-1.1 0-2 .9-2 2v12H6V21zm25 22.8V34h9.8L31 43.8z"
                  />
                </svg>
              </span>
            </span>
          </button>,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <div
    className="custom-input"
  >
    <input
      data-terra-date-input-hidden={true}
      type="hidden"
    />
    <Input
      className="input"
      disabled={false}
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en",
          "messages": null,
          "now": [Function],
          "onError": [Function],
          "textComponent": "span",
          "timeZone": null,
        }
      }
      isInvalid={false}
      name="terra-date-undefined"
      onChange={[Function]}
      required={false}
      type="text"
    >
      <input
        className="form-input input"
        disabled={false}
        intl={
          Object {
            "defaultFormats": Object {},
            "defaultLocale": "en",
            "formatDate": [Function],
            "formatHTMLMessage": [Function],
            "formatMessage": [Function],
            "formatNumber": [Function],
            "formatPlural": [Function],
            "formatRelative": [Function],
            "formatTime": [Function],
            "formats": Object {},
            "formatters": Object {
              "getDateTimeFormat": [Function],
              "getMessageFormat": [Function],
              "getNumberFormat": [Function],
              "getPluralFormat": [Function],
              "getRelativeFormat": [Function],
            },
            "locale": "en",
            "messages": null,
            "now": [Function],
            "onError": [Function],
            "textComponent": "span",
            "timeZone": null,
          }
        }
        name="terra-date-undefined"
        onChange={[Function]}
        required={false}
        type="text"
      />
    </Input>
    <Button
      className="button"
      icon={
        <IconCalendar
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        />
      }
      isBlock={false}
      isCompact={true}
      isDisabled={false}
      isIconOnly={true}
      isReversed={false}
      onClick={[Function]}
      onKeyDown={[Function]}
      refCallback={
        [MockFunction] {
          "calls": Array [
            Array [
              <button
                aria-disabled="false"
                aria-label="Terra.datePicker.openCalendar"
                class="button neutral compact button"
                title="Terra.datePicker.openCalendar"
                type="button"
              >
                <span
                  class="button-label icon-only"
                >
                  <span
                    class="icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="tui-Icon icon icon-svg"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 48 48"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M43 6h-4V2.2C39 1 38 0 36.8 0h-1.7C33.9.1 33 1 33 2.2V6H15V2.2C15 1 14 0 12.8 0h-1.7C9.9.1 9 1 9 2.2V6H5c-1.1 0-2 .9-2 2v38c0 1.1.9 2 2 2h24c1.3-.1 2.5-.6 3.4-1.4l11.2-11.1c.8-.9 1.3-2.1 1.4-3.4V8c0-1.1-.9-2-2-2zM6 21h36v10H30c-1.1 0-2 .9-2 2v12H6V21zm25 22.8V34h9.8L31 43.8z"
                      />
                    </svg>
                  </span>
                </span>
              </button>,
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      text="Terra.datePicker.openCalendar"
      type="button"
      variant="neutral"
    >
      <button
        aria-disabled={false}
        aria-label="Terra.datePicker.openCalendar"
        className="button neutral compact button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        title="Terra.datePicker.openCalendar"
        type="button"
      >
        <span
          className="button-label icon-only"
        >
          <span
            className="icon"
          >
            <IconCalendar
              className="icon-svg"
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
            >
              <IconBase
                ariaLabel={null}
                className="icon-svg"
                focusable="false"
                height="1em"
                isBidi={false}
                isSpin={false}
                viewBox="0 0 48 48"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <svg
                  aria-hidden="true"
                  className="tui-Icon icon icon-svg"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 48 48"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M43 6h-4V2.2C39 1 38 0 36.8 0h-1.7C33.9.1 33 1 33 2.2V6H15V2.2C15 1 14 0 12.8 0h-1.7C9.9.1 9 1 9 2.2V6H5c-1.1 0-2 .9-2 2v38c0 1.1.9 2 2 2h24c1.3-.1 2.5-.6 3.4-1.4l11.2-11.1c.8-.9 1.3-2.1 1.4-3.4V8c0-1.1-.9-2-2-2zM6 21h36v10H30c-1.1 0-2 .9-2 2v12H6V21zm25 22.8V34h9.8L31 43.8z"
                  />
                </svg>
              </IconBase>
            </IconCalendar>
          </span>
        </span>
      </button>
    </Button>
  </div>
</DatePickerInput>
`;

exports[`should render a default date input 1`] = `
<div
  class="custom-input"
>
  <input
    data-terra-date-input-hidden="true"
    type="hidden"
  />
  <input
    class="form-input input"
    intl="[object Object]"
    name="terra-date-undefined"
    type="text"
  />
  <button
    aria-disabled="false"
    aria-label="Terra.datePicker.openCalendar"
    class="button neutral compact button"
    title="Terra.datePicker.openCalendar"
    type="button"
  >
    <span
      class="button-label icon-only"
    >
      <span
        class="icon"
      >
        <svg
          aria-hidden="true"
          class="tui-Icon icon icon-svg"
          focusable="false"
          height="1em"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M43 6h-4V2.2C39 1 38 0 36.8 0h-1.7C33.9.1 33 1 33 2.2V6H15V2.2C15 1 14 0 12.8 0h-1.7C9.9.1 9 1 9 2.2V6H5c-1.1 0-2 .9-2 2v38c0 1.1.9 2 2 2h24c1.3-.1 2.5-.6 3.4-1.4l11.2-11.1c.8-.9 1.3-2.1 1.4-3.4V8c0-1.1-.9-2-2-2zM6 21h36v10H30c-1.1 0-2 .9-2 2v12H6V21zm25 22.8V34h9.8L31 43.8z"
          />
        </svg>
      </span>
    </span>
  </button>
</div>
`;

exports[`should render a default date input with all props 1`] = `
<div
  class="custom-input"
>
  <input
    data-terra-date-input-hidden="true"
    name="date-input"
    type="hidden"
    value="2017-01-01"
  />
  <input
    class="form-input input"
    id="terra-date-input"
    intl="[object Object]"
    name="terra-date-date-input"
    placeholder="MM/DD/YYYY"
    type="text"
    value="01/01/2017"
  />
  <button
    aria-disabled="false"
    aria-label="Terra.datePicker.openCalendar"
    class="button neutral compact button"
    title="Terra.datePicker.openCalendar"
    type="button"
  >
    <span
      class="button-label icon-only"
    >
      <span
        class="icon"
      >
        <svg
          aria-hidden="true"
          class="tui-Icon icon icon-svg"
          focusable="false"
          height="1em"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M43 6h-4V2.2C39 1 38 0 36.8 0h-1.7C33.9.1 33 1 33 2.2V6H15V2.2C15 1 14 0 12.8 0h-1.7C9.9.1 9 1 9 2.2V6H5c-1.1 0-2 .9-2 2v38c0 1.1.9 2 2 2h24c1.3-.1 2.5-.6 3.4-1.4l11.2-11.1c.8-.9 1.3-2.1 1.4-3.4V8c0-1.1-.9-2-2-2zM6 21h36v10H30c-1.1 0-2 .9-2 2v12H6V21zm25 22.8V34h9.8L31 43.8z"
          />
        </svg>
      </span>
    </span>
  </button>
</div>
`;
